client/ayon_houdini/plugins/create/create_generic.py

```diff
@@ -0,0 +1,751 @@
+                    creator_attributes[creator_key] = value
+                    del node_data[key]  # remove from original
+
+            node_data["creator_attributes"] = creator_attributes
+            node_data["publish_attributes"] = publish_attributes
+
+            created_instance = CreatedInstance.from_existing(
+                node_data, self
+            )
+            self._add_instance_to_context(created_instance)
+
+    def update_instances(self, update_list):
+        # Overridden to pass `created_instance` to `self.imprint`
+        for created_inst, changes in update_list:
+            new_values = {
+                key: changes[key].new_value
+                for key in changes.changed_keys
+            }
+            # Update parm templates and values
+            self.imprint(
+                created_inst,
+                new_values,
+                update=True
+            )
+
+    def get_product_name(
+        self,
+        project_name,
+        folder_entity,
+        task_entity,
+        variant,
+        host_name=None,
+        instance=None
+    ):
+        if instance is not None:
+            self.product_type = instance.data["productType"]
+            product_name = super(CreateHoudiniGeneric, self).get_product_name(
+                project_name,
+                folder_entity,
+                task_entity,
+                variant,
+                host_name,
+                instance)
+            self.product_type = "generic"
+            return product_name
+
+        else:
+            return "<-- defined on create -->"
+
+    def create_attribute_def_parms(self,
+                                   node: "hou.OpNode",
+                                   created_instance: CreatedInstance):
+        # We imprint all the attributes into an AYON tab on the node in which
+        # we have a list folder called `attributes` in which we have
+        # - Instance Attributes
+        # - Creator Attributes
+        # - Publish Attributes
+        # With also a separate `advanced` section for specific attributes
+        parm_group = node.parmTemplateGroup()
+
+        # Create default folder parm structure
+        ayon_folder = parm_group.findFolder("AYON")
+        if not ayon_folder:
+            ayon_folder = hou.FolderParmTemplate("folder", "AYON")
+            parm_group.addParmTemplate(ayon_folder)
+
+        # Add publish button
+        publish_button_parm = hou.ButtonParmTemplate(
+            "AYON_self_publish",
+            "Publish",
+            help="Directly publishes only this node and any inputs with a "
+                 "comment entered in a popup prompt. All other instances will "
+                 "be disabled in the publish.",
+            script_callback="from ayon_houdini.api.lib import "
+                            "self_publish; self_publish()",
+            script_callback_language=hou.scriptLanguage.Python,
+        )
+        ayon_folder.addParmTemplate(publish_button_parm)
```

Comment:
This should fix the bug 
```suggestion
        publish_button_parm = parm_group.find("AYON_self_publish")
        if not publish_button_parm:
            publish_button_parm = hou.ButtonParmTemplate(
                "AYON_self_publish",
                "Publish",
                help="Directly publishes only this node and any inputs with a "
                    "comment entered in a popup prompt. All other instances will "
                    "be disabled in the publish.",
                script_callback="from ayon_houdini.api.lib import "
                                "self_publish; self_publish()",
                script_callback_language=hou.scriptLanguage.Python,
            )
            ayon_folder.addParmTemplate(publish_button_parm)
```

Reply:
Also, the above logic always creates self publish button as it  doesn't take the value of the following setting into consideration: 
- `ayon+settings://houdini/general/add_self_publish_button`

For reference, the super class `plugin.HoudiniCreator` already adds the self publish button in `apply_settings` method (which is not overridden in `CreateHoudiniGeneric`)
That's why we may have a publish button duplicate.

https://github.com/ynput/ayon-houdini/blob/c3ab366c047e63881fa4cd95de79284a8605bc11/client/ayon_houdini/api/plugin.py#L295-L297

Reply:
we can also do this which leverage `self.add_publish_button` that is set in `apply_settings`.

```suggestion
        publish_button_parm = parm_group.find("AYON_self_publish")
        if not publish_button_parm and self.add_publish_button:
            publish_button_parm = hou.ButtonParmTemplate(
                "AYON_self_publish",
                "Publish",
                help="Directly publishes only this node and any inputs with a "
                    "comment entered in a popup prompt. All other instances will "
                    "be disabled in the publish.",
                script_callback="from ayon_houdini.api.lib import "
                                "self_publish; self_publish()",
                script_callback_language=hou.scriptLanguage.Python,
            )
            ayon_folder.addParmTemplate(publish_button_parm)
```
But, this has two downsides:
1. it won't add the button until we change any attribute (as it won't be marked as a change on its own)
2. the button will be added at the end of the list because we are using `.addParmTemplate` and folders doesn't support `insertBefore`.
![image](https://github.com/user-attachments/assets/79b0c08a-36a1-4bbe-a67e-37291bb908f6)